src/main/java/com/blamejared/crafttweaker/impl/managers/GenericRecipesManager.java

```diff
@@ -0,0 +1,205 @@
+package com.blamejared.crafttweaker.impl.managers;
+
+import com.blamejared.crafttweaker.CraftTweaker;
+import com.blamejared.crafttweaker.api.CraftTweakerAPI;
+import com.blamejared.crafttweaker.api.annotations.ZenRegister;
+import com.blamejared.crafttweaker.api.data.IData;
+import com.blamejared.crafttweaker.api.item.IIngredient;
+import com.blamejared.crafttweaker.api.managers.IRecipeManager;
+import com.blamejared.crafttweaker.impl.actions.recipes.ActionAddRecipe;
+import com.blamejared.crafttweaker.impl.actions.recipes.generic.ActionRemoveAllGenericRecipes;
+import com.blamejared.crafttweaker.impl.actions.recipes.generic.ActionRemoveGenericRecipeByModId;
+import com.blamejared.crafttweaker.impl.actions.recipes.generic.ActionRemoveGenericRecipeByName;
+import com.blamejared.crafttweaker.impl.actions.recipes.generic.ActionRemoveGenericRecipeByOutput;
+import com.blamejared.crafttweaker.impl.actions.recipes.generic.ActionRemoveGenericRecipeByRegex;
+import com.blamejared.crafttweaker.impl.brackets.RecipeTypeBracketHandler;
+import com.blamejared.crafttweaker.impl.data.MapData;
+import com.blamejared.crafttweaker.impl.recipes.wrappers.WrapperRecipe;
+import com.blamejared.crafttweaker_annotations.annotations.Document;
+import com.google.gson.JsonObject;
+import net.minecraft.item.crafting.IRecipe;
+import net.minecraft.item.crafting.RecipeManager;
+import net.minecraft.util.ResourceLocation;
+import org.openzen.zencode.java.ZenCodeGlobals;
+import org.openzen.zencode.java.ZenCodeType;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.function.BinaryOperator;
+import java.util.function.Function;
+import java.util.stream.Collector;
+import java.util.stream.Collectors;
+
+/**
+ * This recipe manager allows you to perform removal actions over all recipe managers.
+ * This is also the only place where you can specify the serializer type of an addJSON call directly.
+ * You can access this manager by using the `recipes` global keyword.
+ *
+ * @docParam this recipes
+ */
+@ZenRegister
+@Document("vanilla/api/managers/GenericRecipesManager")
+@ZenCodeType.Name("crafttweaker.api.GenericRecipesManager")
+public class GenericRecipesManager {
+    
+    
+    @ZenCodeGlobals.Global("recipes")
+    public static final GenericRecipesManager RECIPES = new GenericRecipesManager();
```

Comment:
Probably unfortunate naming if someone tries to use 1.12.2 syntax in 1.16.5+. Not really a problem, just something we might want to consider for the sake of user support.

Reply:
We discussed it a bit before, basically I don't want to have to not do it because 1.12 did it.

Also the amount of people making this mistake has gone down quite a lot recently